src/parser.c

```diff
@@ -559,3 +559,23 @@ size_t pos_to_str_with_limit(struct pos_array *pos, uint8_t *str, size_t limit)
     }
     return len;
 }
+
+int pos_is_zero(struct pos_array *pos)
+{
+    int length = pos->str_len;
+    if (length <= 0) {
+        LOG(ERROR, "pos_is_zero: string length %d <= 0", length);
+        return CORVUS_ERR;
+    }
+
+    int len = 0;
+    for (size_t i = 0; len < length; i++) {
```

Comment:
`pos_array` is used to represent only one string, consisting of split sub string stored in `items`
``` C
struct pos_array {
    struct pos *items;
    int str_len;                    // str_len == sum(items.len)
    int pos_len;                  // pos_len == len(items)
    int max_pos_size;       // max_pos_size is the actual memory size allocated
};
```